python_modules/dagster/dagster/_daemon/asset_daemon.py

```diff
@@ -342,15 +602,17 @@ def _run_iteration_impl(
             # Evaluation ID will always be monotonically increasing, but will not always
             # be auto-incrementing by 1 once there are multiple AMP evaluations happening in
             # parallel
-            next_evaluation_id = stored_cursor.evaluation_id + 1
+            next_evaluation_id = self._get_next_evaluation_id()
             tick = instance.create_tick(
                 TickData(
-                    instigator_origin_id=FIXED_AUTO_MATERIALIZATION_ORIGIN_ID,
-                    instigator_name=FIXED_AUTO_MATERIALIZATION_INSTIGATOR_NAME,
-                    instigator_type=InstigatorType.AUTO_MATERIALIZE,
+                    instigator_origin_id=instigator_origin_id,
+                    instigator_name=instigator_name,
+                    instigator_type=(
+                        InstigatorType.SENSOR if sensor else InstigatorType.AUTO_MATERIALIZE
+                    ),
                     status=TickStatus.STARTED,
                     timestamp=evaluation_time.timestamp(),
-                    selector_id=FIXED_AUTO_MATERIALIZATION_SELECTOR_ID,
+                    selector_id=instigator_selector_id,
                     auto_materialize_evaluation_id=next_evaluation_id,
```

Comment:
I'm not familiar with the evaluation id.  Does it just uniquely identify the tick?  Where is this used?

I'm wondering what properties we need it to have, but it currently just looks like a sensor-scoped counter.  With the threaded workers, how many ticks can a single worker have concurrently evaluating?

Reply:
Evaluation ID predates AMP moving to use ticks and sensors - it essentially just uniquely identifies the tick, if back-compat and existing evaluations were not a concern it could just use the tick ID, hoping we can combine them together in the future.

Reply:
But for the new AMP sensors, we could just use tick id, or is this some sort of artificial ordering in the UI that we need to do?